### PR TITLE
Get cuda info without nvcc

### LINKED ISSUE #163: Graphic card information in results
It would be good to have the name of the graphic card in addition to the CUDA version currently saved.
With NVIDIA-smi there is the command:

``` bash
nvidia-smi --query-gpu=name --format=csv,noheader
```

so using `subprocess` module (already used for other informations), we can get it easily. Not sure for others though.

### LINKED ISSUE #378: ENH use pynvml to check conda info
I got several issues on platforms where I have a GPU but no access to `nvcc`, as solvers such as `cuml` relies on the fact that we can find a `nvcc`.
I propose to use `pynvml` to check the GPU info: this will be more robust and this will also allow to access to the name of the GPU, which is more informative (knowing if we have a `V100` or a `Titan`).

### PR DESCRIPTION
**Pull request by tanglef**
Fixes #378 and Fixes #163 by collection GPU name and cuda version
This doesn't use `pynvml` (which is "a wrapper around NVML library" that does, more quickly, what the smi thinner wrapper `nvidia-smi -q -x` does)

### CODE REVIEW

File: benchopt/utils/sys_info.py [hunk 1]
```diff
@@ -26,13 +26,14 @@ def _get_processor_name():
...
+    if which("nvidia-smi") is None:
         return None
-    command = ["nvcc", "--version"]
+    command = ["nvidia-smi", "-q", "-x"]
     out = subprocess.check_output(command).strip().decode("utf-8")
-    out = out.splitlines()[-1]  # take only last line
-    return out
+    version = re.search('<cuda_version>(.*)</cuda_version>', out).group(1)
+    name = re.search('<product_name>(.*)</product_name>', out).group(1)
+    return f"{name}: cuda_{version}"
```
Comment (tomMoral) [inline benchopt/utils/sys_info.py:36]:
```suggestion
    try:
        version = re.search('<cuda_version>(.*)</cuda_version>', out).group(1)
        name = re.search('<product_name>(.*)</product_name>', out).group(1)
        return f"{name}: cuda_{version}"
    except AttributeError:
        import warnings
        warngings.warn("Found `nvidia-smi` but could not parse cuda version "
                                  "or device name.")
        return None
```

**Review by tomMoral — APPROVED** [review]
Nice, it is indeed pretty neat. Just a small nitpick to better protect in case the output is not parse correctly, as we don't want to lose benchmark results due to bad parsing in the end.

File: benchopt/utils/sys_info.py (hunk 1 above)
Comment (tanglef) [reply benchopt/utils/sys_info.py:36]: yes, didn't think about this case. Thanks

File: benchopt/utils/sys_info.py [hunk 2]
```diff
@@ -26,13 +26,20 @@ def _get_processor_name():
 
 
 def _get_cuda_version():
```
Comment (agramfort) [inline benchopt/utils/sys_info.py:28]: I would now make this function public as we use it in benchmark code

Comment (tomMoral) [reply benchopt/utils/sys_info.py:28]: Good call

**Comment by tanglef** [comment]
Doc build fails because of `lasso`'s snapml call to `_get_cuda_version()`

File: benchopt/utils/sys_info.py [hunk 3]
```diff
@@ -25,14 +25,21 @@ def _get_processor_name():
...
-    out = out.splitlines()[-1]  # take only last line
-    return out
+    try:
+        version = re.search('<cuda_version>(.*)</cuda_version>', out).group(1)
+        name = re.search('<product_name>(.*)</product_name>', out).group(1)
+        return f"{name}: cuda_{version}"
+    except AttributeError:
+        import warnings
+        warnings.warn("Found `nvidia-smi` but could not parse cuda version "
+                      "or device name.")
```
Comment (tomMoral) [inline benchopt/utils/sys_info.py:41]:
```suggestion
        warnings.warn(
            "Could not parse cuda version or device name from `nvidia-smi`."
         )
```

**Pull request merged**